security/pkg/k8s/tokenreview/k8sauthn.go

```diff
@@ -16,22 +16,56 @@ package tokenreview
 
 import (
 	"context"
+	"encoding/base64"
+	"encoding/json"
 	"fmt"
 	"strings"
 
 	k8sauth "k8s.io/api/authentication/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 
-	"istio.io/istio/pkg/jwt"
+	"istio.io/pkg/env"
+	"istio.io/pkg/log"
 )
 
-const (
+var (
 	// The default audience for SDS trustworthy JWT. This is to make sure that the CSR requests
 	// contain the JWTs intended for Citadel.
-	DefaultAudience = "istio-ca"
+	DefaultAudience = env.RegisterStringVar("TOKEN_AUDIENCE", "istio-ca", "Audience to check in accepted JWTW tokens")
+	RequireAudience = env.RegisterBoolVar("REQUIRE_AUDIENCE", false, "Reject tokens without audience. If false, K8S token will be accepted")
 )
 
+type jwtPayload struct {
+	// Aud is the expected audience, defaults to istio-ca - but is based on istiod.yaml configuration.
+	// If set to a different value - use the value defined by istiod.yaml. Env variable can
```

Comment:
So the logic is, the default value is "istio-ca", but overridable by istiod.yaml and Env variable? Overriding in two places is confusing. Why don't we just override it in env var?

Reply:
Yes, comment is confusing. Changed to make it more clear. 

It is just the env var - will be added to MeshConfig after API review, if needed. 
The common pattern for audience is to use the base URL of the authority - part of review we can discuss if this is something
that should be exposed to the user.